pkg/kubelet/kubelet_pods.go

```diff
@@ -1635,7 +1635,15 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 				}
 				// Secondary IP is not set #105320
 				if len(hostIPs) == 2 && len(s.PodIPs) == 1 {
-					s.PodIPs = append(s.PodIPs, v1.PodIP{IP: hostIPs[1].String()})
+					currHostIPIsIPv6 := utilnet.ParseIPSloppy(s.PodIPs[0].IP).To4() == nil
+					newHostIPIsIPv6 := utilnet.ParseIPSloppy(hostIPs[1].String()).To4() == nil
+					if currHostIPIsIPv6 != newHostIPIsIPv6 {
```

Comment:
we have already some utils 
```suggestion
					if err, ok utilnet.IsDualStackIPStrings(hostIPs) ; err == nil && ok {
```

Reply:
Thank you!. I think `err == nil && ok` should not be handled together. For `e2e != nil `, I use a different warning message about invalid host IP.